client/init.js

```diff
@@ -87,17 +88,45 @@ addEventListener('load', function() {
 
 	function startWatchingPosition() {
 		navigator.geolocation.watchPosition(function(position) {
-			try {
-				comms.send({
-					event: 'friend-updated',
-					position: position
-				})
-			} catch (e) {
-				console.error(e)
-			}
+			sendMessage({
+				event: 'friend-updated',
+				position: position
+			})
 		})
 	}
 
+	function setupHiddenEvent(){
+		var visibility = new Visibility()
+		function handleVisibilityChange(){
+			var hiding = document[visibility.hidden]
+			if (hiding){
+				timeoutId = window.setTimeout(function(){
+					sendMessage({
+						event: 'friend-updated',
+						stale: true,
+						hiding: true
+					})
+				}, timeout)
+			}
+			else
+				window.clearTimeout(timeoutId)
+			sendMessage({
+				event: 'friend-updated',
```

Comment:
I'd give this a different event. Something like friend-activity-state-changed or something less unwieldy. This way, everyone doesn't need to check whether properties are found or not on the message.

Basically, each message should act like a well known type, where if you get a message with the type you care about, it's properties should be accessible. Maybe we should make types for them and share them somehow...